src/core/crypto.h

```diff
@@ -39,6 +39,12 @@ typedef struct QUIC_CRYPTO {
     //
     BOOLEAN CertValidationPending : 1;
 
+    //
+    // Indicates Resumption ticket validation is under validation asynchronously
+    //
+    QUIC_SERVER_APP_TICKET_VALIDATION_STATUS TicketValidationStatus : 2;
```

Comment:
enums (which are signed) have issues (or at least in my experience they do) with bitfields. I think it might be better to simply have two BOOLEANS, "pending" and "previous failed" (or something like that).

Reply:
Yeah, actually single flag is enough to do same thing, but it makes implementation complicated and need to use `if (Pending) {}` for **rejecting** ticket by second call. which is not intuitive.